### PR TITLE
New Database Deletion Flow

### PR DESCRIPTION
**Pull request by ItzNotABug**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/routes/console/project-[project]/databases/database-[database]/settings/+page.svelte [hunk 1]
```diff
@@ -93,14 +93,14 @@
             </div>
 
             <p>
-                The database will be permanently deleted, including all data associated with this
-                team. This action is irreversible.
+                The database will be permanently deleted, including all the collections within it.
```
Comment (heyCarla) [inline src/routes/console/project-[project]/databases/database-[database]/settings/+page.svelte:96]:
```suggestion
                The database will be permanently deleted, including all collections within it.
```

**Comment by ItzNotABug** [comment]
Todo: Add message that the database deletion will also delete its associated backups.

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte [hunk 2]
```diff
@@ -30,20 +97,120 @@
             trackError(error, Submit.DatabaseDelete);
         }
     };
+
+    /* reset data on modal close */
+    $: if (!showDelete) {
+        databaseName = '';
```
Comment (ernstmul) [inline src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:103]: Shouldn't the databaseName also be reset to its initial value `null`?

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte [hunk 3]
```diff
@@ -30,20 +97,120 @@
...
+
+<style>
+    :global(.max-height-dialog dialog) {
+        max-height: 650px !important;
+    }
+
+    :global(.no-inner-borders .table) {
+        box-shadow: unset !important;
+        border-radius: unset !important;
+    }
```
Comment (ernstmul) [inline src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:215]: Are these `!important`'s really needed?

File: src/routes/(console)/project-[project]/databases/database-[database]/+page.ts [hunk 4]
```diff
@@ -3,22 +3,26 @@ import { sdk } from '$lib/stores/sdk';
...
 import { CARD_LIMIT, Dependencies } from '$lib/constants';
+import { collections as collectionsStore } from './store';
 
 export const load: PageLoad = async ({ params, url, route, depends }) => {
     depends(Dependencies.COLLECTIONS);
     const page = getPage(url);
     const limit = getLimit(url, route, CARD_LIMIT);
     const view = getView(url, route, View.Grid);
     const offset = pageToOffset(page, limit);
+    const collections = await sdk.forProject.databases.listCollections(params.database, [
```
Comment (TorstenDittmann) [inline src/routes/(console)/project-[project]/databases/database-[database]/+page.ts:14]: can we check and see, in the layout of collections we also fetch documents - lets remove it there

Comment (TorstenDittmann) [reply src/routes/(console)/project-[project]/databases/database-[database]/+page.ts:14]: actually lets sync with design on this, we shouldnt be fetching collections without deleting 👍🏻

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte [hunk 5]
```diff
@@ -30,20 +97,120 @@
...
+
+<style>
+    :global(.max-height-dialog dialog) {
+        max-height: 650px;
+    }
+
+    :global(.no-inner-borders .table) {
+        box-shadow: unset;
+        border-radius: unset;
+    }
```
Comment (TorstenDittmann) [inline src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:215]: We should have an option for this, can you check the Table props?

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte [hunk 6]
```diff
@@ -30,20 +97,120 @@
...
+                Delete
+            </Button>
+        </svelte:fragment>
+    </Modal>
+</div>
+
+<style>
+    :global(.max-height-dialog dialog) {
+        max-height: 650px;
+    }
```
Comment (TorstenDittmann) [inline src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:210]: why do we need that?

Comment (ItzNotABug) [reply src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:210]: the collections list can increase and that would increase the modal height too iirc.

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte [hunk 7]
```diff
@@ -4,14 +4,81 @@
...
+                };
+            });
+
+            collectionItems = [...collectionItems, ...(await Promise.all(collectionPromises))];
+        } catch (err) {
+            error = true;
+        } finally {
+            isLoadingDocumentsCount = false;
+        }
+    }
```
Comment (TorstenDittmann) [inline src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:80]: I am not sure we can do this, there can be an infinite amount of collections - therefore we should probably implement backend for this.

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte (hunk 5 above)
Comment (ItzNotABug) [reply src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:215]: I don't think this even makes a difference, I see no changes with or without it, wonder why I added it back then. As per props, I see `noMargins` and `noStyles`. Wouldn't use `noStyles` on this.

File: src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte (hunk 7 above)
Comment (ItzNotABug) [reply src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte:80]: But wont the queries help? This is sort of paginated behaviour. wdyt?

Comment (ItzNotABug) [reply src/routes/(console)/project-[project]/databases/database-[database]/delete.svelte]: Done.

**Review by ernstmul — APPROVED** [review]

**Review by ernstmul — APPROVED** [review]

**Pull request merged**